graphql/resolvers.go

```diff
@@ -971,42 +971,41 @@ func (r *patchResolver) ID(ctx context.Context, obj *restModel.APIPatch) (string
 	return *obj.Id, nil
 }
 
-func (r *patchResolver) Status(ctx context.Context, obj *restModel.APIPatch) (string, error) {
-	failedAndAbortedStatuses := append(evergreen.TaskFailureStatuses, evergreen.TaskAborted)
-	opts := data.TaskFilterOptions{
-		Statuses:        failedAndAbortedStatuses,
-		FieldsToProject: []string{task.DisplayStatusKey},
-	}
-	tasks, _, err := r.sc.FindTasksByVersion(*obj.Id, opts)
+func (r *queryResolver) Patch(ctx context.Context, id string) (*restModel.APIPatch, error) {
+	patch, err := r.sc.FindPatchById(id)
 	if err != nil {
-		return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
+		return nil, InternalServerError.Send(ctx, err.Error())
 	}
-	status := *obj.Status
-	if len(obj.ChildPatches) > 0 {
-		for _, cp := range obj.ChildPatches {
-			// add the child patch tasks to tasks so that we can consider their status
-			childPatchTasks, _, err := r.sc.FindTasksByVersion(*cp.Id, opts)
-			if err != nil {
-				return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
-			}
-			tasks = append(tasks, childPatchTasks...)
+
+	if evergreen.IsFinishedPatchStatus(*patch.Status) {
```

Comment:
I would have expected this code to look like https://github.com/evergreen-ci/evergreen/blob/c7d7bb97a84b3db534e5149d33a1070dbd86d183/graphql/resolvers.go#L974-L1010 if this is a revert, but I don't see `IsFinishedPatchStatus` there.

I don't need to be blocking though, since I don't have context on this change.

Reply:
I added this in to try and improve the performance a little over the original implementation so it would only try to resolve  the patch status when the patch has completed.